### PR TITLE
1913 extref converting and validation

### PR DESCRIPTION
**Pull request by mrix**
* Add `ExtReferenceConverter`
* Add `extref` validation

### CODE REVIEW

**Comment by hairmare** [comment]
Hi @mrix 

We recently merged some changes in #212 that show what kind of collections are allowed to be used as reference in the schema output of ``extref`` fields. Does this patch take already take this into account? I see the new validator component that was written, but it's unclear if it does more than just validate that an URL is an URL. Could you clarify that? I'll try and find some time to test this before merging, it might take a bit though (today is meeting day).

Cheers
Lucas

**Comment by mrix** [comment]
Hi @hairmare. No `ExtrefValidator` validates only URL format and route.
I will add new constraint option `allowedCollections` and checking reference validity.

**Comment by mrix** [comment]
@hairmare please check `ExtReferenceValidator`

**Comment by narcoticfresh** [comment]
@hairmare
hairmare raised the suspicion that my PR #214 could interfere with this.. but looking over the changed files it doesn't seem to conflict..

@mrix 
do you see any problems in PR #214 for this PR? would you be ok if we first merge #214 and you merge develop again?

**Comment by mrix** [comment]
I tried to merge #214 into my branch. There is no conflicts. All tests are passed successfully.

File: src/Graviton/CoreBundle/Resources/definition/Module.json [hunk 1]
```diff
@@ -134,7 +134,8 @@
         "title": "Language-Link",
         "description": "Link to language document.",
         "exposeAs": "$ref",
-        "required": true
+        "required": true,
+        "collection": ["App"]
```
Comment (hairmare) [inline src/Graviton/CoreBundle/Resources/definition/Module.json]: This should reference the ``Language`` collection...

Comment (mrix) [inline src/Graviton/CoreBundle/Resources/definition/Module.json]: Fixed.

**Pull request merged**